chatsky/script/core/context.py

```diff
@@ -110,155 +113,129 @@ class Context(BaseModel):
-        self.labels[last_index + 1] = label
+            if main is None:
+                raise ValueError(f"Context with id {id} not found in the storage!")
+            crt_at, upd_at, fw_data = main
+            objected = storage.serializer.loads(fw_data)
+            instance = cls(id=id, framework_data=objected, turns=turns, misc=misc)
+            instance._created_at, instance._updated_at, instance._storage = crt_at, upd_at, storage
+            return instance
+
+    async def store(self) -> None:
+        if self._storage is not None:
+            self._updated_at = time_ns()
+            byted = self._storage.serializer.dumps(self.framework_data)
+            await launch_coroutines(
+                [
+                    self._storage.update_main_info(self.primary_id, self._created_at, self._updated_at, byted),
+                    self.turns.store(),
+                    self.misc.store(),
+                ],
+                self._storage.is_asynchronous,
+            )
+        else:
+            raise RuntimeError(f"{type(self).__name__} is not attached to any context storage!")
 
     def clear(
         self,
         hold_last_n_indices: int,
-        field_names: Union[Set[str], List[str]] = {"requests", "responses", "labels"},
+        field_names: Union[Set[str], List[str]] = {"turns"},
     ):
-        """
-        Delete all records from the `requests`/`responses`/`labels` except for
-        the last `hold_last_n_indices` turns.
-        If `field_names` contains `misc` field, `misc` field is fully cleared.
-
-        :param hold_last_n_indices: Number of last turns to keep.
-        :param field_names: Properties of :py:class:`~.Context` to clear.
-            Defaults to {"requests", "responses", "labels"}
-        """
         field_names = field_names if isinstance(field_names, set) else set(field_names)
-        if "requests" in field_names:
-            for index in list(self.requests)[:-hold_last_n_indices]:
-                del self.requests[index]
-        if "responses" in field_names:
-            for index in list(self.responses)[:-hold_last_n_indices]:
-                del self.responses[index]
+        if "turns" in field_names:
+            del self.turns[:-hold_last_n_indices]
         if "misc" in field_names:
             self.misc.clear()
-        if "labels" in field_names:
-            for index in list(self.labels)[:-hold_last_n_indices]:
-                del self.labels[index]
         if "framework_data" in field_names:
             self.framework_data = FrameworkData()
 
+    async def delete(self) -> None:
+        if self._storage is not None:
+            await self._storage.delete_main_info(self.primary_id)
+        else:
+            raise RuntimeError(f"{type(self).__name__} is not attached to any context storage!")
+
+    @property
+    def labels(self) -> Dict[int, NodeLabel2Type]:
+        return {id: turn.label for id, turn in self.turns.items()}
```

Comment:
So accessing `ctx.labels` would load all turns from db?
I think we should do something similar to `ctx_dict` here to have `labels` return a dict that loads labels on demand.
Alternatively, we can make a method for getting labels that would accept a key/slice and return an iterable since now the `labels` dict is supposed to be view-only (addition of new labels is done via turns).